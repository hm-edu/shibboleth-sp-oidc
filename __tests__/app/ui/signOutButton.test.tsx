import { test, expect, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import SignOutButton from '@/app/ui/signOutButton';

test('SignOutButton', () => {
  vi.mock('@/auth', () => {
    return { signOut: vi.fn(() => Promise.resolve()) };
  });
  render(<SignOutButton />);
  expect(screen.getByRole('button')).toBeDefined();
  expect(screen.getByRole('button')).toMatchSnapshot();
  fireEvent.click(screen.getByText(/Abmelden/));
});
