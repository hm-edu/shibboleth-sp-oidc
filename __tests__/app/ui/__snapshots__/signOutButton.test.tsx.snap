// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignOutButton > should match snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-sbfmij-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="submit"
>
  Abmelden
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`SignOutButton 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-sbfmij-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="submit"
>
  Abmelden
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
